### ISSUE TITLE
luarocks needs writeable /usr/local

### ISSUE THREAD

**Issue by shawnacscott** [Mashape/homebrew-kong#39]
While trying to install kong via this formula, the installation would fail at the luarocks openssl step with:
```
Error: Your user does not have write permissions in /usr/local
-- you may want to run as a privileged user or use your local tree with --local.
```

Since Homebrew no longer requires write permissions for `/usr/local`, I would expect this formula not to rely on having those permissions either.

**Comment by MuraliGK** [comment]
I'm also facing the same issue. Any workaround..?

**Comment by thibaultcha** [comment]
Thanks for reporting. We would welcome a contribution to fix this issue.

**Comment by shawnacscott** [comment]
@thibaultcha Thanks! I'm not very familiar with the DSL Homebrew formulas like this use. If you have any helpful resources for getting started with them, I'd love to see them. Otherwise, I can do some research on my own when I have time. 

@MuraliGK the workaround I used was to make `/usr/local` writeable (`sudo chown -R $(whoami) /usr/local`), go through the installation instructions, then change the permissions back to what they were before (for me, `sudo chown root:wheel /usr/local`). I don't know that this won't cause problems with the installation later, but it seems to be working for me so far.

**Comment by nihalgonsalves** [comment]
I have the same issue, not sure how to fix it.

@shawnacscott That works for now, thanks! Don't forget the `-R` flag when changing back to root:wheel!

**Comment by shawnacscott** [comment]
@nihalgonsalves The Homebrew messaging around permissions does not require the `-R` flag:

```
Homebrew no longer needs to have ownership of /usr/local. If you wish you can
return /usr/local to its default ownership with:
  sudo chown root:wheel /usr/local
```

Additionally, I didn't find `sudo chown -R $(whoami) /usr/local/Cellar` to be necessary.

**Comment by jason-riddle** [comment]
Here's a quick, no permission change, manual fix for the time being.

```
Error: Your user does not have write permissions in /usr/local
-- you may want to run as a privileged user or use your local tree with --local.
```

So just add the `--local` flag.

```
$ cd /Library/Caches/Homebrew/kong--git
$ luarocks make OPENSSL_DIR=/usr/local/opt/openssl --local
```

Finally, copy the binary to somewhere in your path and you should be able to use kong just fine.

```
$ cp bin/kong /opt/bin
```

So all that needs to happen is this line needs to be changed.

https://github.com/Mashape/homebrew-kong/blob/88608d7/Formula/kong.rb#L26

```diff
diff --git a/Formula/kong.rb b/Formula/kong.rb
index 6e19b48..8c5e4e0 100644
--- a/Formula/kong.rb
+++ b/Formula/kong.rb
@@ -23,7 +23,7 @@ class Kong < Formula
   conflicts_with "cassandra", :because => "Kong only supports Cassandra 2.1/2.2"
 
   def install
-    system "luarocks make OPENSSL_DIR=#{Formula['openssl'].opt_prefix}"
+    system "luarocks make OPENSSL_DIR=#{Formula['openssl'].opt_prefix} --local"
     bin.install "bin/kong"
   end
 end
```

**Comment by jason-riddle** [comment]
Alright I spoke too soon, there appear to be some other issues.

```
$ ./bin/kong
./bin/kong:11: module 'cassandra.socket' not found:
	no field package.preload['cassandra.socket']
	no file '/usr/local/Cellar/openresty/1.11.2.1/site/lualib/cassandra/socket.lua'
	no file '/usr/local/Cellar/openresty/1.11.2.1/site/lualib/cassandra/socket/init.lua'
	no file '/usr/local/Cellar/openresty/1.11.2.1/lualib/cassandra/socket.lua'
	no file '/usr/local/Cellar/openresty/1.11.2.1/lualib/cassandra/socket/init.lua'
	no file './cassandra/socket.lua'
	no file '/usr/local/Cellar/openresty/1.11.2.1/luajit/share/luajit-2.1.0-beta2/cassandra/socket.lua'
	no file '/usr/local/share/lua/5.1/cassandra/socket.lua'
	no file '/usr/local/share/lua/5.1/cassandra/socket/init.lua'
	no file '/usr/local/Cellar/openresty/1.11.2.1/luajit/share/lua/5.1/cassandra/socket.lua'
	no file '/usr/local/Cellar/openresty/1.11.2.1/luajit/share/lua/5.1/cassandra/socket/init.lua'
	no file '/usr/local/Cellar/openresty/1.11.2.1/site/lualib/cassandra/socket.so'
	no file '/usr/local/Cellar/openresty/1.11.2.1/lualib/cassandra/socket.so'
	no file './cassandra/socket.so'
	no file '/usr/local/lib/lua/5.1/cassandra/socket.so'
	no file '/usr/local/Cellar/openresty/1.11.2.1/luajit/lib/lua/5.1/cassandra/socket.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
	no file '/usr/local/Cellar/openresty/1.11.2.1/site/lualib/cassandra.so'
	no file '/usr/local/Cellar/openresty/1.11.2.1/lualib/cassandra.so'
	no file './cassandra.so'
	no file '/usr/local/lib/lua/5.1/cassandra.so'
	no file '/usr/local/Cellar/openresty/1.11.2.1/luajit/lib/lua/5.1/cassandra.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
```

**Comment by ulfmagnetics** [comment]
The issue with `cassandra.socket` issue is a separate one, apparently: https://github.com/Mashape/homebrew-kong/issues/38. It's not related to the `/usr/local` permissions issue.

**Comment by caledhwa** [comment]
Wow, this makes the `homebrew` install actually unusable. Isn't this a full-on blocking bug with this install method? Since this is the only MacOS install solution in the docs, isn't this a bit concerning?

**Comment by thibaultcha** [comment]
This should have been fixed with #43, thanks for the report.
